File: leetcode2/1easy/강병호/Q1021.py
class Solution:
    def removeOuterParentheses(self, s: str) -> str:
        stack = []
        result = []

        for c in s:
            if c == '(':
                if stack: # 스택이 비어있지 않으면 last아님
                    result.append(c)
                stack.append(c)
            else:
                stack.pop()
                if stack : # 스택이 비어있지 않으면 last아님
                    result.append(c)
        return ''.join(result)

File: leetcode2/1easy/김대원/Q1021.java
class Solution {
    public String removeOuterParentheses(String s) {
        boolean flag = false;
        Stack<Character> stack = new Stack<>();
        char[] chars = s.toCharArray();
        StringBuilder answer = new StringBuilder();
        for (char ch : chars) {
            if (ch == '(') {
                stack.push(ch);
                if (stack.size() >= 2) {
                    answer.append("(");
                    flag = true;
                }
            } else {
                stack.pop();
                if (stack.isEmpty()) flag = false;
                if (flag) answer.append(")");
            }
        }

        return answer.toString();
    }
}